server/src/main/java/org/elasticsearch/transport/RemoteClusterAware.java

```diff
@@ -181,12 +186,36 @@ protected RemoteClusterAware(Settings settings) {
      * {@link TransportAddress#META_ADDRESS} and their configured address will be used as the hostname for the generated discovery node.
      */
     protected static Map<String, Tuple<String, List<Supplier<DiscoveryNode>>>> buildRemoteClustersDynamicConfig(Settings settings) {
-        Stream<Setting<List<String>>> allConcreteSettings = REMOTE_CLUSTERS_SEEDS.getAllConcreteSettings(settings);
+        final Map<String, Tuple<String, List<Supplier<DiscoveryNode>>>> remoteSeeds =
+                buildRemoteClustersDynamicConfig(settings, REMOTE_CLUSTERS_SEEDS);
+        final Map<String, Tuple<String, List<Supplier<DiscoveryNode>>>> searchRemoteSeeds =
+                buildRemoteClustersDynamicConfig(settings, SEARCH_REMOTE_CLUSTERS_SEEDS);
+        // sort the intersection for predictable output order
+        final NavigableSet<String> intersection =
+                new TreeSet<>(Arrays.asList(
+                        searchRemoteSeeds.keySet().stream().filter(s -> remoteSeeds.keySet().contains(s)).sorted().toArray(String[]::new)));
```

Comment:
would accepting only one or the other simplify things a bit? I guess there is no reason to use the new setting and the deprecated one at the same time?

Reply:
I noticed that but it seems that you can still use the two settings at the same time, say one for a cluster and the other for another cluster. Is that correct?

Reply:
Yes; and that strikes me as okay so that we can support existing configurations, and users can add or migrate clusters to the new settings one at a time.